### ISSUE TITLE
Set UID and GID in Ansible config

### ISSUE THREAD

**Issue by fnoah** [spantaleev/matrix-docker-ansible-deploy#268]
This playbook uses UID and GID 991 for synapse. Since another user / group was already using them on my server, I changed the Synapse UID and GID to 741 in `vars.yml`:
```
matrix_user_uid: 741
matrix_user_gid: 741
```

Once I started the installation procedure I got the following error:
```
TASK [matrix-synapse : Generate initial Synapse config and signing key] *******************************************************************************************************************************************
fatal: [matrix.<domain>]: FAILED! => {"changed": true, "cmd": ["docker", "run", "--rm", "--name=matrix-config", "--user=741:741", "--cap-drop=ALL", "-v", "/matrix/synapse/config:/data", "-e", "SYNAPSE_CONFIG_PATH=/data/homeserver.yaml", "-e", "SYNAPSE_SERVER_NAME=matrix.<domain>", "-e", "SYNAPSE_REPORT_STATS=no", "matrixdotorg/synapse:v1.3.1", "generate"], "delta": "0:00:02.291808", "end": "2019-09-23 16:02:13.541668", "msg": "non-zero return code", "rc": 1, "start": "2019-09-23 16:02:11.249860", "stderr": "chown: /data: Operation not permitted\nCreating log config /data/matrix.<domain>.log.config\nTraceback (most recent call last):\n  File \"/start.py\", line 243, in <module>\n    main(sys.argv, os.environ)\n  File \"/start.py\", line 182, in main\n    return run_generate_config(environ, ownership)\n  File \"/start.py\", line 153, in run_generate_config\n    subprocess.check_output([\"chown\", ownership, data_dir])\n  File \"/usr/local/lib/python3.7/subprocess.py\", line 395, in check_output\n    **kwargs).stdout\n  File \"/usr/local/lib/python3.7/subprocess.py\", line 487, in run\n    output=stdout, stderr=stderr)\nsubprocess.CalledProcessError: Command '['chown', '991:991', '/data']' returned non-zero exit status 1.", "stderr_lines": ["chown: /data: Operation not permitted", "Creating log config /data/matrix.<domain>.log.config", "Traceback (most recent call last):", "  File \"/start.py\", line 243, in <module>", "    main(sys.argv, os.environ)", "  File \"/start.py\", line 182, in main", "    return run_generate_config(environ, ownership)", "  File \"/start.py\", line 153, in run_generate_config", "    subprocess.check_output([\"chown\", ownership, data_dir])", "  File \"/usr/local/lib/python3.7/subprocess.py\", line 395, in check_output", "    **kwargs).stdout", "  File \"/usr/local/lib/python3.7/subprocess.py\", line 487, in run", "    output=stdout, stderr=stderr)", "subprocess.CalledProcessError: Command '['chown', '991:991', '/data']' returned non-zero exit status 1."], "stdout": "", "stdout_lines": []}
```

What's notable is the following: `Command '['chown', '991:991', '/data']`

Some script still seems to be using the default UID:GID 991 instead of the `matrix_user_uid` and `matrix_user_gid` variables.

Is there a reason that UID:GID 991 must be used or should it be possible to set those values manually in my `vars.yml` config?

**Comment by spantaleev** [comment]
It should be possible to override it using `matrix_user_uid` and `matrix_user_gid`.
It's all largely untested though, so it may hit some troubles.

I think we had a bug for the "Generate initial Synapse config and signing key" and it was defaulting to `991:991` (which is what Synapse uses by default). I've attempted a fix at 73c90e9.

Please confirm whether it works for you and whether you're hitting any other bugs.

**Comment by fnoah** [comment]
At least in my use case, this seems to work. Thank you.
